### DISCUSSION TITLE
Way to keep extended directus in CVS

### DISCUSSION THREAD

**Discussion by enstain** [directus/directus#2721] [Archive]
How do you deal with extensions in Directus in terms of code version control? It must be a pain not to have extensions in the same codebase as the core, but the core has .gitignore that states that it ignores everything in the extensions folder. 

So what's the use case for production?

Do you keep extensions in a separate repository and then pull all of them when you do next deployment to the server?  Or do you clone Directus core into new repository and then deploy this whole thing to the server? But what about updates? I find it pretty nice to keep Directus core pure as in original repo, so I can always pull origin to update Directus to the next version without touching my custom code. So is it really about multiple repositories then?

**Comment by rijkvanzanten** [comment]
> Do you keep extensions in a separate repository and then pull all of them when you do next deployment to the server?

That's how I do it, yes

> Or do you clone Directus core into new repository and then deploy this whole thing to the server?

No that would be a fork and even if you don't make changes to the core codebase it ends up being annoying

>  I find it pretty nice to keep Directus core pure as in original repo, so I can always pull origin to update Directus to the next version without touching my custom code.

Yes, that's the idea here

> So is it really about multiple repositories then?

I think so!

---

This is all a little simpler to handle in Docker, as you can mount the volume that holds the extensions and handle the source control for that mounted directory separately from your docker instance that can update through Docker(Hub).

@WoLfulus is the path for extensions (next to uploads) something we can make configurable in api-next? Being able to have the Directus source stuff next to the extensions instead of having extensions inside the Directus folders should negate all of this as well

**Reply by enstain** [reply]
> Being able to have the Directus source stuff next to the extensions instead of having extensions inside the Directus folders should negate all of this as well

Sounds like an awesome plan) Thanks for clarifications on the official practice)

**Reply by WoLfulus** [reply]
on api-next I'm going for composer packages, so you'll be able to install them with `composer require`
